### PR TITLE
Python cleanups/v4

### PR DESCRIPTION
**Pull request by inashivb**
Previous PR: https://github.com/OISF/suricata/pull/9082

Changes since v3:
- Remove unnecessary shebang
- remove exec permission during install on scripts

### CODE REVIEW

**Comment by victorjulien** [comment]
Windows and Mac CI unhappy, can you check what is going on?

**Comment by inashivb** [comment]
> Windows and Mac CI unhappy, can you check what is going on?

hmm from what I'm able to understand, one needs execute perm to copy files to a dir. So, I'm now wondering why is it working wit Linux..

File: doc/userguide/unix-socket.rst [hunk 1]
```diff
@@ -42,11 +42,13 @@ example to write custom scripts:
...
 ---------------------------------
-You may need to install ``suricatasc`` if you have not done so, running the following command from python/suricatasc
+Runnable script for suricatasc is available in `python/bin` directory of suricata. You can
+run it with the following commands.
 
 ::
 
-  sudo python setup.py install
+  cd python
+  sudo ./bin/suricatasc
```
Comment (jasonish) [inline doc/userguide/unix-socket.rst:51]: These changes are correct, but this does need a rewrite.  More of a mental note for now.

File: python/Makefile.am [hunk 2]
```diff
@@ -20,10 +20,10 @@ EXTRA_DIST = $(LIBS) bin suricata/config/defaults.py
...
 
 install-exec-local:
-	install -d -m 0755 "$(DESTDIR)$(prefix)/lib/suricata/python/suricata/config"
-	install -d -m 0755 "$(DESTDIR)$(prefix)/lib/suricata/python/suricata/ctl"
-	install -d -m 0755 "$(DESTDIR)$(prefix)/lib/suricata/python/suricata/sc"
-	install -d -m 0755 "$(DESTDIR)$(prefix)/lib/suricata/python/suricatasc"
+	install -d -m 0644 "$(DESTDIR)$(prefix)/lib/suricata/python/suricata/config"
+	install -d -m 0644 "$(DESTDIR)$(prefix)/lib/suricata/python/suricata/ctl"
+	install -d -m 0644 "$(DESTDIR)$(prefix)/lib/suricata/python/suricata/sc"
+	install -d -m 0644 "$(DESTDIR)$(prefix)/lib/suricata/python/suricatasc"
```
Comment (jasonish) [inline python/Makefile.am:26]: As there are directories, they need the `0755` permissions.

File: python/Makefile.am [hunk 3]
```diff
@@ -20,10 +20,10 @@ EXTRA_DIST = $(LIBS) bin suricata/config/defaults.py
...
 	install -d -m 0755 "$(DESTDIR)$(prefix)/bin"
 	for src in $(LIBS); do \
 		install $(srcdir)/$$src "$(DESTDIR)$(prefix)/lib/suricata/python/$$src"; \
```
Comment (jasonish) [inline python/Makefile.am:29]: It is here we should add `-m 0644`.

**Review by jasonish — CHANGES_REQUESTED** [review]
Changes requested. Doc is OK, but I may work on something new for that section.

File: python/Makefile.am (hunk 2 above)
Comment (inashivb) [reply python/Makefile.am:26]: yeah. but, any idea why this works with Linux?

File: python/Makefile.am (hunk 3 above)
Comment (inashivb) [reply python/Makefile.am:29]: ok thanks

**Pull request closed without merging**